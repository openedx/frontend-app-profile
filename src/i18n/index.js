import { messages as headerMessages } from '@edx/frontend-component-header';
import { messages as footerMessages } from '@edx/frontend-component-footer';
import { messages as paragonMessages } from '@openedx/paragon';
import arMessages from './messages/ar.json';
import deMessages from './messages/de.json';
import dedeCAMessages from './messages/de_DE.json';
import es419Messages from './messages/es_419.json';
import faIRMessages from './messages/fa_IR.json';
import frCAMessages from './messages/fr_CA.json';
import itMessages from './messages/it.json';
import ititCAMessages from './messages/it_IT.json';
import frMessages from './messages/fr.json';
import hiMessages from './messages/hi.json';
import ptMessages from './messages/pt.json';
import ptptCAMessages from './messages/pt_PT.json';
import ruMessages from './messages/ru.json';
import ukMessages from './messages/uk.json';
import zhcnMessages from './messages/zh_CN.json';
// no need to import en messages-- they are in the defaultMessage field

const appMessages = {
  ar: arMessages,
  'es-419': es419Messages,
  'fa-ir': faIRMessages,
  fr: frMessages,
  'zh-cn': zhcnMessages,
  pt: ptMessages,
  it: itMessages,
  de: deMessages,
  hi: hiMessages,
  'fr-ca': frCAMessages,
  ru: ruMessages,
  uk: ukMessages,
  'de-de': dedeCAMessages,
  'it-it': ititCAMessages,
  'pt-pt': ptptCAMessages,
};

export default [
  headerMessages,
  footerMessages,
  paragonMessages,
  appMessages,
];
