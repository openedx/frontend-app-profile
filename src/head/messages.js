import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  'profile.page.title': {
    id: 'profile.page.title',
    defaultMessage: 'Profile | {siteName}',
    description: 'Title tag',
  },
});

export default messages;
