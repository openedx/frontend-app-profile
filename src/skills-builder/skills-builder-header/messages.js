import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  skillsBuilderHeaderTitle: {
    id: 'skills.builder.header.title',
    defaultMessage: 'Skills Builder',
    description: 'Title for the Skills Builder feature',
  },
  skillsBuilderHeaderSubheading: {
    id: 'skills.builder.header.subheading',
    defaultMessage: 'Let edX be your guide',
    description: 'Subheading to the Skills Builder title in the header component',
  },
});

export default messages;
