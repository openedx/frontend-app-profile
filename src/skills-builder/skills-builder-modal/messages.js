import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  /* Modal Action Row Buttons */
  goBackButton: {
    id: 'go.back.button',
    defaultMessage: 'Go Back',
    description: 'Button that sends the user to the previous step in the skills builder.',
  },
  nextStepButton: {
    id: 'next.step.button',
    defaultMessage: 'Next Step',
    description: 'Button that sends the user to the next step in the skills builder.',
  },
  exitButton: {
    id: 'exit.button',
    defaultMessage: 'Exit',
    description: 'Button that exits the Skills Builder.',
  },
  selectPreferences: {
    id: 'select.preferences',
    defaultMessage: 'Select preferences',
    description: 'The first step of the Skills Builder for selecting a goal, a current job/occupation, and career interests',
  },
  reviewResults: {
    id: 'review.results',
    defaultMessage: 'Review results',
    description: 'The second step of the Skills Builder for rendering results from learner input',
  },
});

export default messages;
