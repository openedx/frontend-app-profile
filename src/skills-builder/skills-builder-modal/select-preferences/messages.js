import { defineMessages } from '@edx/frontend-platform/i18n';

const messages = defineMessages({
  skillsBuilderDescription: {
    id: 'skills.builder.description',
    defaultMessage: 'Find the right courses and programs that help you reach your goals.',
    description: 'Description of what the Skills Builder seeks to accomplish',
  },
  learningGoalPrompt: {
    id: 'learning.goal.prompt',
    defaultMessage: 'First, tell us what you want to achieve',
    description: 'Prompts the user to select their current goal.',
  },
  selectLearningGoal: {
    id: 'select.learning.goal',
    defaultMessage: 'Select a goal',
    description: 'Placeholder text for the goal selection component.',
  },
  learningGoalStartCareer: {
    id: 'learning.goal.start_career',
    defaultMessage: 'I want to start my career',
    description: 'Selected by user if their goal is to start their career.',
  },
  learningGoalAdvanceCareer: {
    id: 'learning.goal.advance_career',
    defaultMessage: 'I want to advance my career',
    description: 'Selected by user if their goal is to advance their career.',
  },
  learningGoalChangeCareer: {
    id: 'learning.goal.change_career',
    defaultMessage: 'I want to change careers',
    description: 'Selected by user if their goal is to change careers.',
  },
  learningGoalSomethingNew: {
    id: 'learning.goal.something.new',
    defaultMessage: 'I want to learn something new',
    description: 'Selected by user if their goal is to learn something new.',
  },
  learningGoalSomethingElse: {
    id: 'learning.goal.something.else',
    defaultMessage: 'Something else',
    description: 'Selected by user if their goal is not described by the other choices.',
  },
  jobTitlePrompt: {
    id: 'job.title.prompt',
    defaultMessage: 'Next, search and select your current job title',
    description: 'Prompts the user to select their current job title or occupation.',
  },
  studentCheckboxPrompt: {
    id: 'student.checkbox.prompt',
    defaultMessage: 'I\'m a student',
    description: 'Label text for the corresponding checkbox',
  },
  currentlyLookingCheckboxPrompt: {
    id: 'currently.looking.checkbox.prompt',
    defaultMessage: 'I\'m currently looking for work',
    description: 'Label text for the corresponding checkbox',
  },
  careerInterestPrompt: {
    id: 'career.interest.prompt',
    defaultMessage: 'What careers are you interested in?',
    description: 'Prompts the user to select careers they are interested in pursuing.',
  },
  removeCareerInterestButtonAltText: {
    id: 'career.interest.remove.button.alt.text',
    defaultMessage: 'Remove career interest: ',
  },
});

export default messages;
