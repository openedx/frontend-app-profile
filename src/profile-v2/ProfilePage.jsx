import React, {
  useEffect, useState, useContext, useCallback,
} from 'react';
import PropTypes from 'prop-types';
import { useDispatch, useSelector } from 'react-redux';
import { sendTrackingLogEvent } from '@edx/frontend-platform/analytics';
import { ensureConfig, getConfig } from '@edx/frontend-platform';
import { AppContext } from '@edx/frontend-platform/react';
import { useIntl } from '@edx/frontend-platform/i18n';
import { Alert, Hyperlink } from '@openedx/paragon';
import classNames from 'classnames';
import {
  fetchProfile,
  saveProfilePhoto,
  deleteProfilePhoto,
} from './data/actions';
import ProfileAvatar from './forms/ProfileAvatar';
import Certificates from './Certificates';
import DateJoined from './DateJoined';
import UserCertificateSummary from './UserCertificateSummary';
import UsernameDescription from './UsernameDescription';
import PageLoading from './PageLoading';
import { profilePageSelector } from './data/selectors';
import messages from './ProfilePage.messages';
import withParams from '../utils/hoc';
import { useIsOnMobileScreen, useIsOnTabletScreen } from './data/hooks';

ensureConfig(['CREDENTIALS_BASE_URL', 'LMS_BASE_URL'], 'ProfilePage');

const ProfilePage = ({ params }) => {
  const dispatch = useDispatch();
  const intl = useIntl();
  const context = useContext(AppContext);
  const {
    requiresParentalConsent,
    dateJoined,
    yearOfBirth,
    courseCertificates,
    name,
    profileImage,
    savePhotoState,
    isLoadingProfile,
    photoUploadError,
  } = useSelector(profilePageSelector);

  const [viewMyRecordsUrl, setViewMyRecordsUrl] = useState(null);
  const isMobileView = useIsOnMobileScreen();
  const isTabletView = useIsOnTabletScreen();

  useEffect(() => {
    const { CREDENTIALS_BASE_URL } = context.config;
    if (CREDENTIALS_BASE_URL) {
      setViewMyRecordsUrl(`${CREDENTIALS_BASE_URL}/records`);
    }

    dispatch(fetchProfile(params.username));
    sendTrackingLogEvent('edx.profile.viewed', {
      username: params.username,
    });
  }, [dispatch, params.username, context.config]);

  const handleSaveProfilePhoto = useCallback((formData) => {
    dispatch(saveProfilePhoto(context.authenticatedUser.username, formData));
  }, [dispatch, context.authenticatedUser.username]);

  const handleDeleteProfilePhoto = useCallback(() => {
    dispatch(deleteProfilePhoto(context.authenticatedUser.username));
  }, [dispatch, context.authenticatedUser.username]);

  const isYOBDisabled = () => {
    const currentYear = new Date().getFullYear();
    const isAgeOrNotCompliant = !yearOfBirth || ((currentYear - yearOfBirth) < 13);
    return isAgeOrNotCompliant && getConfig().COLLECT_YEAR_OF_BIRTH !== 'true';
  };

  const isAuthenticatedUserProfile = () => params.username === context.authenticatedUser.username;

  const isBlockVisible = (blockInfo) => isAuthenticatedUserProfile()
      || (!isAuthenticatedUserProfile() && Boolean(blockInfo));

  const renderViewMyRecordsButton = () => {
    if (!(viewMyRecordsUrl && isAuthenticatedUserProfile())) {
      return null;
    }

    return (
      <Hyperlink
        className={classNames(
          'btn btn-brand btn-rounded font-weight-normal px-4 py-0625rem text-nowrap',
          { 'btn-sm': isMobileView },
        )}
        target="_blank"
        showLaunchIcon={false}
        destination={viewMyRecordsUrl}
      >
        {intl.formatMessage(messages['profile.viewMyRecords'])}
      </Hyperlink>
    );
  };

  const renderPhotoUploadErrorMessage = () => (
    photoUploadError && (
    <div className="row">
      <div className="col-md-4 col-lg-3">
        <Alert variant="danger" dismissible={false} show>
          {photoUploadError.userMessage}
        </Alert>
      </div>
    </div>
    )
  );

  return (
    <div className="profile-page">
      {isLoadingProfile ? (
        <PageLoading srMessage={intl.formatMessage(messages['profile.loading'])} />
      ) : (
        <>
          <div
            className={classNames(
              'profile-page-bg-banner bg-primary d-md-block align-items-center py-4rem h-100 w-100',
              { 'px-4.5': isMobileView },
              { 'px-75rem': !isMobileView },
            )}
          >
            <div className="col container-fluid w-100 h-100 bg-white py-0 px-25rem rounded-75">
              <div className="col h-100 w-100 py-4 px-0 justify-content-start g-15rem">
                <div
                  className={classNames([
                    'row-auto d-flex flex-wrap align-items-center h-100 w-100 justify-content-start g-15rem',
                    isMobileView || isTabletView ? 'flex-column' : 'flex-row',
                  ])}
                >
                  <ProfileAvatar
                    className="col p-0"
                    src={profileImage.src}
                    isDefault={profileImage.isDefault}
                    onSave={handleSaveProfilePhoto}
                    onDelete={handleDeleteProfilePhoto}
                    savePhotoState={savePhotoState}
                    isEditable={isAuthenticatedUserProfile() && !requiresParentalConsent}
                  />
                  <div
                    className={classNames([
                      'col h-100 w-100 m-0 p-0',
                      isMobileView || isTabletView
                        ? 'd-flex flex-column justify-content-center align-items-center'
                        : 'justify-content-start align-items-start',
                    ])}
                  >
                    <p className={classNames([
                      'row m-0 font-weight-bold text-truncate text-primary-500',
                      isMobileView ? 'h4' : 'h3',
                    ])}
                    >
                      {params.username}
                    </p>
                    {isBlockVisible(name) && (
                    <p className={classNames([
                      'row pt-2 text-gray-800 font-weight-normal m-0',
                      isMobileView ? 'h5' : 'p',
                    ])}
                    >
                      {name}
                    </p>
                    )}
                    <div className={classNames(
                      'row pt-2 m-0',
                      isMobileView
                        ? 'd-flex justify-content-center align-items-center flex-column'
                        : 'g-1rem',
                    )}
                    >
                      <DateJoined date={dateJoined} />
                      <UserCertificateSummary count={courseCertificates.length} />
                    </div>
                  </div>
                  <div className={classNames([
                    'p-0 ',
                    isMobileView || isTabletView ? 'col d-flex justify-content-center' : 'col-auto',
                  ])}
                  >
                    {renderViewMyRecordsButton()}
                  </div>
                </div>
                <div className="row-auto d-flex align-items-center h-100 w-100 justify-content-start m-0 pt-4">
                  {isYOBDisabled() && <UsernameDescription />}
                </div>
              </div>
              <div className="ml-auto">
                {renderPhotoUploadErrorMessage()}
              </div>
            </div>
          </div>
          <div className="col container-fluid d-inline-flex px-75rem pt-4rem pb-6 h-100 w-100 align-items-start justify-content-center g-3rem">
            {isBlockVisible(courseCertificates.length) && (
              <Certificates
                certificates={courseCertificates}
                formId="certificates"
              />
            )}
          </div>
        </>
      )}
    </div>
  );
};

ProfilePage.propTypes = {
  params: PropTypes.shape({
    username: PropTypes.string.isRequired,
  }).isRequired,
};

export default withParams(ProfilePage);
