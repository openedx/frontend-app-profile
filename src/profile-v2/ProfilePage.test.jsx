import { getConfig } from '@edx/frontend-platform';
import * as analytics from '@edx/frontend-platform/analytics';
import { AppContext } from '@edx/frontend-platform/react';
import { configure as configureI18n, IntlProvider } from '@edx/frontend-platform/i18n';
import { render } from '@testing-library/react';
import React from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import messages from '../i18n';
import ProfilePage from './ProfilePage';
import loadingApp from './__mocks__/loadingApp.mockStore';
import viewOwnProfile from './__mocks__/viewOwnProfile.mockStore';
import viewOtherProfile from './__mocks__/viewOtherProfile.mockStore';

const mockStore = configureMockStore([thunk]);

const storeMocks = {
  loadingApp,
  viewOwnProfile,
  viewOtherProfile,
};
const requiredProfilePageProps = {
  fetchUserAccount: () => {},
  fetchProfile: () => {},
  params: { username: 'staff' },
};

// Mock language cookie
Object.defineProperty(global.document, 'cookie', {
  writable: true,
  value: `${getConfig().LANGUAGE_PREFERENCE_COOKIE_NAME}=en`,
});

jest.mock('@edx/frontend-platform/auth', () => ({
  configure: () => {},
  getAuthenticatedUser: () => null,
  fetchAuthenticatedUser: () => null,
  getAuthenticatedHttpClient: jest.fn(),
  AUTHENTICATED_USER_CHANGED: 'user_changed',
}));

jest.mock('@edx/frontend-platform/analytics', () => ({
  configure: () => {},
  identifyAnonymousUser: jest.fn(),
  identifyAuthenticatedUser: jest.fn(),
  sendTrackingLogEvent: jest.fn(),
}));

configureI18n({
  loggingService: { logError: jest.fn() },
  config: {
    ENVIRONMENT: 'production',
    LANGUAGE_PREFERENCE_COOKIE_NAME: 'yum',
  },
  messages,
});

beforeEach(() => {
  analytics.sendTrackingLogEvent.mockReset();
});

const ProfilePageWrapper = ({
  contextValue, store, params,
}) => (
  <AppContext.Provider
    value={contextValue}
  >
    <IntlProvider locale="en">
      <Provider store={store}>
        <ProfilePage {...requiredProfilePageProps} params={params} />
      </Provider>
    </IntlProvider>
  </AppContext.Provider>
);

ProfilePageWrapper.defaultProps = {
  params: { username: 'staff' },
};

ProfilePageWrapper.propTypes = {
  contextValue: PropTypes.shape({}).isRequired,
  store: PropTypes.shape({}).isRequired,
  params: PropTypes.shape({}),
};

describe('<ProfilePage />', () => {
  describe('Renders correctly in various states', () => {
    it('app loading', () => {
      const contextValue = {
        authenticatedUser: { userId: null, username: null, administrator: false },
        config: getConfig(),
      };
      const component = <ProfilePageWrapper contextValue={contextValue} store={mockStore(storeMocks.loadingApp)} />;
      const { container: tree } = render(component);
      expect(tree).toMatchSnapshot();
    });

    it('viewing own profile', () => {
      const contextValue = {
        authenticatedUser: { userId: 123, username: 'staff', administrator: true },
        config: getConfig(),
      };
      const component = <ProfilePageWrapper contextValue={contextValue} store={mockStore(storeMocks.viewOwnProfile)} />;
      const { container: tree } = render(component);
      expect(tree).toMatchSnapshot();
    });

    it('viewing other profile with all fields', () => {
      const contextValue = {
        authenticatedUser: { userId: 123, username: 'staff', administrator: true },
        config: getConfig(),
      };

      const component = (
        <ProfilePageWrapper
          contextValue={contextValue}
          store={mockStore({
            ...storeMocks.viewOtherProfile,
            profilePage: {
              ...storeMocks.viewOtherProfile.profilePage,
              account: {
                ...storeMocks.viewOtherProfile.profilePage.account,
                name: 'user',
                country: 'EN',
                bio: 'bio',
                courseCertificates: ['course certificates'],
                levelOfEducation: 'some level',
                languageProficiencies: ['some lang'],
                socialLinks: ['twitter'],
                timeZone: 'time zone',
                accountPrivacy: 'all_users',
              },
            },
          })}
          match={{ params: { username: 'verified' } }} // Override default match
        />
      );
      const { container: tree } = render(component);
      expect(tree).toMatchSnapshot();
    });

    it('without credentials service', () => {
      const config = getConfig();
      config.CREDENTIALS_BASE_URL = '';

      const contextValue = {
        authenticatedUser: { userId: 123, username: 'staff', administrator: true },
        config: getConfig(),
      };
      const component = (
        <ProfilePageWrapper
          contextValue={contextValue}
          store={mockStore(storeMocks.viewOwnProfile)}
        />
      );
      const { container: tree } = render(component);
      expect(tree).toMatchSnapshot();
    });
  });

  describe('handles analytics', () => {
    it('calls sendTrackingLogEvent when mounting', () => {
      const contextValue = {
        authenticatedUser: { userId: 123, username: 'staff', administrator: true },
        config: getConfig(),
      };
      render(
        <ProfilePageWrapper
          contextValue={contextValue}
          store={mockStore(storeMocks.loadingApp)}
          params={{ username: 'test-username' }}
        />,
      );

      expect(analytics.sendTrackingLogEvent.mock.calls.length).toBe(1);
      expect(analytics.sendTrackingLogEvent.mock.calls[0][0]).toEqual('edx.profile.viewed');
      expect(analytics.sendTrackingLogEvent.mock.calls[0][1]).toEqual({
        username: 'test-username',
      });
    });
  });
});
