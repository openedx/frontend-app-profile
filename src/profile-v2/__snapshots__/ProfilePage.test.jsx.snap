// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfilePage /> Renders correctly in various states app loading 1`] = `
<div>
  <div
    class="profile-page"
  >
    <div>
      <div
        class="d-flex justify-content-center align-items-center flex-column height-50vh"
      >
        <div
          class="spinner-border text-primary"
          role="status"
        >
          <span
            class="sr-only"
          >
            Profile loading...
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ProfilePage /> Renders correctly in various states viewing other profile with all fields 1`] = `
<div>
  <div
    class="profile-page"
  >
    <div
      class="profile-page-bg-banner bg-primary d-md-block align-items-center px-75rem py-4rem h-100 w-100"
    >
      <div
        class="col container-fluid w-100 h-100 bg-white py-0 px-25rem rounded-75"
      >
        <div
          class="col h-100 w-100 py-4 px-0 justify-content-start g-15rem"
        >
          <div
            class="row-auto d-flex flex-wrap align-items-center h-100 w-100 justify-content-start g-15rem"
          >
            <div
              class="profile-avatar-wrap position-relative"
            >
              <div
                class="profile-avatar rounded-circle bg-light"
              >
                <iconmock
                  aria-hidden="true"
                  class="text-muted"
                  focusable="false"
                  role="img"
                  viewbox="0 0 24 24"
                />
              </div>
              <form
                enctype="multipart/form-data"
              >
                <input
                  accept=".jpg, .jpeg, .png"
                  class="d-none form-control-file"
                  id="photo-file"
                  name="file"
                  type="file"
                />
              </form>
            </div>
            <div
              class="col justify-content-start align-items-start h-100 w-100 m-0 p-0"
            >
              <h1
                class="row h3 m-0 font-weight-bold text-truncate text-primary-500"
              >
                staff
              </h1>
              <p
                class="row pt-2 text-gray-800 font-weight-normal m-0"
              >
                user
              </p>
              <div
                class="row pt-2 m-0 g-1rem"
              >
                <span
                  class="small mb-0 text-gray-800"
                >
                  Member since 
                  <span
                    class="font-weight-bold"
                  >
                     
                    2017
                     
                  </span>
                </span>
                <span
                  class="small m-0 text-gray-800"
                >
                  <span
                    class="font-weight-bold"
                  >
                     
                    0
                     
                  </span>
                   certifications
                </span>
              </div>
            </div>
            <div
              class="col-auto p-0"
            >
              <a
                class="pgn__hyperlink default-link standalone-link btn btn-brand btn-rounded font-weight-normal px-4 py-0625rem text-nowrap"
                href="http://localhost:18150/records"
                rel="noopener noreferrer"
                target="_blank"
              >
                View My Records
              </a>
            </div>
          </div>
          <div
            class="row-auto d-flex align-items-center h-100 w-100 justify-content-start m-0 pt-4"
          >
            <p
              class="text-gray-800 font-weight-normal m-0"
            >
              Your learner records information is only visible to you. Only your username is visible to others on localhost.
            </p>
          </div>
        </div>
        <div
          class="ml-auto"
        />
      </div>
    </div>
    <div
      class="col container-fluid d-inline-flex px-75rem pt-4rem pb-6 h-100 w-100 align-items-start justify-content-center g-3rem"
    >
      <div>
        <div
          class="col justify-content-start align-items-start g-5rem p-0"
        >
          <div
            class="col align-self-stretch height-2625rem justify-content-start align-items-start p-0"
          >
            <h2
              class="font-weight-bold text-primary-500 m-0"
            >
              Your certificates
            </h2>
          </div>
          <div
            class="col justify-content-start align-items-start pt-2 p-0"
          >
            <p
              class="font-weight-normal text-gray-800 m-0 p-0"
            >
              Your learner records information is only visible to you. Only your username is visible to others on localhost.
            </p>
          </div>
        </div>
        <div
          class="pt-5"
        >
          You don't have any certificates yet.
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ProfilePage /> Renders correctly in various states viewing own profile 1`] = `
<div>
  <div
    class="profile-page"
  >
    <div
      class="profile-page-bg-banner bg-primary d-md-block align-items-center px-75rem py-4rem h-100 w-100"
    >
      <div
        class="col container-fluid w-100 h-100 bg-white py-0 px-25rem rounded-75"
      >
        <div
          class="col h-100 w-100 py-4 px-0 justify-content-start g-15rem"
        >
          <div
            class="row-auto d-flex flex-wrap align-items-center h-100 w-100 justify-content-start g-15rem"
          >
            <div
              class="profile-avatar-wrap position-relative"
            >
              <div
                class="profile-avatar rounded-circle bg-light"
              >
                <div
                  class="profile-avatar-menu-container"
                >
                  <div
                    class="pgn__dropdown pgn__dropdown-light dropdown"
                    data-testid="dropdown"
                  >
                    <button
                      aria-expanded="false"
                      aria-haspopup="true"
                      class="dropdown-toggle btn btn-primary"
                      type="button"
                    >
                      Change
                    </button>
                  </div>
                </div>
                <img
                  alt="profile avatar"
                  class="w-100 h-100 d-block rounded-circle overflow-hidden object-fit-cover"
                  data-hj-suppress="true"
                  src="http://localhost:18000/media/profile-images/d2a9bdc2ba165dcefc73265c54bf9a20_500.jpg?v=1552495012"
                />
              </div>
              <form
                enctype="multipart/form-data"
              >
                <input
                  accept=".jpg, .jpeg, .png"
                  class="d-none form-control-file"
                  id="photo-file"
                  name="file"
                  type="file"
                />
              </form>
            </div>
            <div
              class="col justify-content-start align-items-start h-100 w-100 m-0 p-0"
            >
              <h1
                class="row h3 m-0 font-weight-bold text-truncate text-primary-500"
              >
                staff
              </h1>
              <p
                class="row pt-2 text-gray-800 font-weight-normal m-0"
              >
                Lemon Seltzer
              </p>
              <div
                class="row pt-2 m-0 g-1rem"
              >
                <span
                  class="small mb-0 text-gray-800"
                >
                  Member since 
                  <span
                    class="font-weight-bold"
                  >
                     
                    2017
                     
                  </span>
                </span>
                <span
                  class="small m-0 text-gray-800"
                >
                  <span
                    class="font-weight-bold"
                  >
                     
                    1
                     
                  </span>
                   certifications
                </span>
              </div>
            </div>
            <div
              class="col-auto p-0"
            >
              <a
                class="pgn__hyperlink default-link standalone-link btn btn-brand btn-rounded font-weight-normal px-4 py-0625rem text-nowrap"
                href="http://localhost:18150/records"
                rel="noopener noreferrer"
                target="_blank"
              >
                View My Records
              </a>
            </div>
          </div>
          <div
            class="row-auto d-flex align-items-center h-100 w-100 justify-content-start m-0 pt-4"
          />
        </div>
        <div
          class="ml-auto"
        />
      </div>
    </div>
    <div
      class="col container-fluid d-inline-flex px-75rem pt-4rem pb-6 h-100 w-100 align-items-start justify-content-center g-3rem"
    >
      <div>
        <div
          class="col justify-content-start align-items-start g-5rem p-0"
        >
          <div
            class="col align-self-stretch height-2625rem justify-content-start align-items-start p-0"
          >
            <h2
              class="font-weight-bold text-primary-500 m-0"
            >
              Your certificates
            </h2>
          </div>
          <div
            class="col justify-content-start align-items-start pt-2 p-0"
          >
            <p
              class="font-weight-normal text-gray-800 m-0 p-0"
            >
              Your learner records information is only visible to you. Only your username is visible to others on localhost.
            </p>
          </div>
        </div>
        <div
          class="col"
        >
          <div
            class="row align-items-center pt-5 g-3rem"
          >
            <div
              class="col-auto d-flex align-items-center p-0"
            >
              <div
                class="col certificate p-4 border-light-400 bg-light-200 w-100 h-100"
              >
                <div
                  class="certificate-type-illustration"
                  style="background-image: url(icon/mock/path);"
                />
                <div
                  class="d-flex flex-column position-relative p-0 width-19625rem"
                >
                  <div
                    class="w-100 color-black"
                  >
                    <p
                      class="small mb-0 font-weight-normal"
                    >
                      Verified Certificate
                    </p>
                    <h4
                      class="m-0 color-black"
                    >
                      edX Demonstration Course
                    </h4>
                    <p
                      class="small mb-0"
                    >
                      From
                    </p>
                    <h5
                      class="mb-0 color-black"
                    >
                      edX
                    </h5>
                    <p
                      class="small mb-0"
                    >
                      Completed on 
                      3/4/2019
                    </p>
                  </div>
                  <div
                    class="pt-3"
                  >
                    <a
                      class="pgn__hyperlink default-link standalone-link btn btn-primary btn-rounded font-weight-normal px-4 py-0625rem"
                      href="http://www.example.com/"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      View Certificate
                    </a>
                  </div>
                  <p
                    class="small mb-0 pt-3"
                  >
                    Credential ID 
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ProfilePage /> Renders correctly in various states without credentials service 1`] = `
<div>
  <div
    class="profile-page"
  >
    <div
      class="profile-page-bg-banner bg-primary d-md-block align-items-center px-75rem py-4rem h-100 w-100"
    >
      <div
        class="col container-fluid w-100 h-100 bg-white py-0 px-25rem rounded-75"
      >
        <div
          class="col h-100 w-100 py-4 px-0 justify-content-start g-15rem"
        >
          <div
            class="row-auto d-flex flex-wrap align-items-center h-100 w-100 justify-content-start g-15rem"
          >
            <div
              class="profile-avatar-wrap position-relative"
            >
              <div
                class="profile-avatar rounded-circle bg-light"
              >
                <div
                  class="profile-avatar-menu-container"
                >
                  <div
                    class="pgn__dropdown pgn__dropdown-light dropdown"
                    data-testid="dropdown"
                  >
                    <button
                      aria-expanded="false"
                      aria-haspopup="true"
                      class="dropdown-toggle btn btn-primary"
                      type="button"
                    >
                      Change
                    </button>
                  </div>
                </div>
                <img
                  alt="profile avatar"
                  class="w-100 h-100 d-block rounded-circle overflow-hidden object-fit-cover"
                  data-hj-suppress="true"
                  src="http://localhost:18000/media/profile-images/d2a9bdc2ba165dcefc73265c54bf9a20_500.jpg?v=1552495012"
                />
              </div>
              <form
                enctype="multipart/form-data"
              >
                <input
                  accept=".jpg, .jpeg, .png"
                  class="d-none form-control-file"
                  id="photo-file"
                  name="file"
                  type="file"
                />
              </form>
            </div>
            <div
              class="col justify-content-start align-items-start h-100 w-100 m-0 p-0"
            >
              <h1
                class="row h3 m-0 font-weight-bold text-truncate text-primary-500"
              >
                staff
              </h1>
              <p
                class="row pt-2 text-gray-800 font-weight-normal m-0"
              >
                Lemon Seltzer
              </p>
              <div
                class="row pt-2 m-0 g-1rem"
              >
                <span
                  class="small mb-0 text-gray-800"
                >
                  Member since 
                  <span
                    class="font-weight-bold"
                  >
                     
                    2017
                     
                  </span>
                </span>
                <span
                  class="small m-0 text-gray-800"
                >
                  <span
                    class="font-weight-bold"
                  >
                     
                    1
                     
                  </span>
                   certifications
                </span>
              </div>
            </div>
            <div
              class="col-auto p-0"
            />
          </div>
          <div
            class="row-auto d-flex align-items-center h-100 w-100 justify-content-start m-0 pt-4"
          />
        </div>
        <div
          class="ml-auto"
        />
      </div>
    </div>
    <div
      class="col container-fluid d-inline-flex px-75rem pt-4rem pb-6 h-100 w-100 align-items-start justify-content-center g-3rem"
    >
      <div>
        <div
          class="col justify-content-start align-items-start g-5rem p-0"
        >
          <div
            class="col align-self-stretch height-2625rem justify-content-start align-items-start p-0"
          >
            <h2
              class="font-weight-bold text-primary-500 m-0"
            >
              Your certificates
            </h2>
          </div>
          <div
            class="col justify-content-start align-items-start pt-2 p-0"
          >
            <p
              class="font-weight-normal text-gray-800 m-0 p-0"
            >
              Your learner records information is only visible to you. Only your username is visible to others on localhost.
            </p>
          </div>
        </div>
        <div
          class="col"
        >
          <div
            class="row align-items-center pt-5 g-3rem"
          >
            <div
              class="col-auto d-flex align-items-center p-0"
            >
              <div
                class="col certificate p-4 border-light-400 bg-light-200 w-100 h-100"
              >
                <div
                  class="certificate-type-illustration"
                  style="background-image: url(icon/mock/path);"
                />
                <div
                  class="d-flex flex-column position-relative p-0 width-19625rem"
                >
                  <div
                    class="w-100 color-black"
                  >
                    <p
                      class="small mb-0 font-weight-normal"
                    >
                      Verified Certificate
                    </p>
                    <h4
                      class="m-0 color-black"
                    >
                      edX Demonstration Course
                    </h4>
                    <p
                      class="small mb-0"
                    >
                      From
                    </p>
                    <h5
                      class="mb-0 color-black"
                    >
                      edX
                    </h5>
                    <p
                      class="small mb-0"
                    >
                      Completed on 
                      3/4/2019
                    </p>
                  </div>
                  <div
                    class="pt-3"
                  >
                    <a
                      class="pgn__hyperlink default-link standalone-link btn btn-primary btn-rounded font-weight-normal px-4 py-0625rem"
                      href="http://www.example.com/"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      View Certificate
                    </a>
                  </div>
                  <p
                    class="small mb-0 pt-3"
                  >
                    Credential ID 
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
